modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c

```diff
@@ -568,7 +568,15 @@ ngx_http_dyups_init_process(ngx_cycle_t *cycle)
     timer->log = cycle->log;
     timer->data = dmcf;
 
-    ngx_add_timer(timer, dmcf->read_msg_timeout);
+    /*
+     * when init process, break up timer, in case of shpool->mutex compete
```

Comment:
Can you describe in detail the problem you encountered?

Reply:
if every worker process  start timer at the same time,  `ngx_shmtx_lock` (in timer's handler)  may  mutex compete in all processes ,  there ere hang in the process of not getting mutex.  When  break up the start time of timer ,  avoid  mutex compete in all  processes.